### PR TITLE
Simplify parsing

### PR DESCRIPTION
**Pull request by darthscsi**

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -1294,6 +1306,20 @@ sub-element in the bundle.
 Components of reference and analog type are ignored, as are any reference or
 analog types within the component (as they cannot be connected to).
 
+### Alternate Syntax
+
+Is invalid may also be specified by keyword.
```
Comment (seldridge) [inline spec.md:1311]:
```suggestion
`is invalid`{.firrtl} may also be specified by keyword.
```

**Review by seldridge — APPROVED** [review]
This will greatly simplify parsing. I had originally thought of going the direction of adding sigils to identifiers like LLVM/MLIR, e.g., `%foo <= %bar`. However, this also works and is much less of a diff.

**Comment by mwachs5** [comment]
makes sense! I agree with the bikeshed color suggested :)

File: spec.md [hunk 2]
```diff
@@ -1294,6 +1307,23 @@ sub-element in the bundle.
...
+`is invalid`.{.firrtl} may also be specified by keyword.
+
+``` firrtl
+module MyModule :
+  input in: {flip a: UInt, b: UInt}
+  output out: {flip a: UInt, b: UInt}
+  wire w: {flip a: UInt, b: UInt}
+  invalidated in
+  ; equivalent to "in is invalid"
+  invalidated out
```
[the remark this answers is not in the document]
Comment (darthscsi) [reply spec.md:1321]: the original was declarative , so I was sticking with that.  Don't care much though.

File: spec.md [hunk 3]
```diff
@@ -1070,6 +1070,19 @@ connected to the left-hand side field.  Conversely, if the i'th field is
...
+### Alternate Syntax
+
+Connects may also be specified by keyword.  This form is identical to the `<=` 
+form in operand order
+
+``` firrtl
+module MyModule :
+  input myinput: UInt
+  output myoutput: UInt
+  connect myoutput, myinput
```
[the remark this answers is not in the document]
Comment (darthscsi) [reply spec.md:1082]: This adds a new separator which I wasn't fond of, but it doesn't matter too much.

File: spec.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (darthscsi) [reply spec.md]: As you wish.

**Pull request merged**